tests/devices/qubit/test_initialize_state.py

```diff
@@ -125,3 +125,5 @@ def test_create_initial_state_with_sparse(self):
         assert not sp.sparse.issparse(state), "State should be converted to dense."
         # The single 1 should be at index 1
         assert state[0, 1] == 1.0
+        assert qml.math.get_interface(state) == "numpy"
+        assert state.shape == (2, 2)
```

Comment:
debugging shows that the shape is `(1, 2, 2)` right now. Weird

Reply:
@albi3ro I tried the same test suite but changed the input vec into usual dense vec. And I saw the final shape is still (1, 2, 2). If this is indeed expected instead of bug, I'll say we could just modify the asserts here accordingly?

Reply:
:scream:

Reply:
Oh I got it. I tested with directly `.toarray()` after the first line. And then of course `csr_matrix` will return a (1, 4) array.

Reply:
Resolved.